src/test/unit_tests/test_namespace_fs.js

```diff
@@ -472,6 +474,22 @@ mocha.describe('namespace_fs', function() {
 
         });
 
+        mocha.it('delete dir object without last slash - /x/y/z/', async function() {
+            const source = buffer_utils.buffer_to_read_stream(data);
+            await upload_object(ns_tmp, upload_bkt, upload_key_4, dummy_object_sdk, source);
+            await delete_object(ns_tmp, upload_bkt, '/x/y/z', dummy_object_sdk);
```

Comment:
I think it would be easier to read with additional variable:
```suggestion
           const partial_dir_3 = dir_3.slice(0, -1); // the path without the last slash
            await delete_object(ns_tmp, upload_bkt, partial_dir_3, dummy_object_sdk);
```